plugins/ipc-rules/ipc-rules.cpp

```diff
@@ -76,7 +76,12 @@ class ipc_rules_t : public wf::plugin_interface_t,
 
     wf::ipc::method_callback get_view_info = [=] (wf::json_t data)
     {
-        auto id = wf::ipc::json_get_uint64(data, "id");
+        auto id = wf::ipc::json_get_optional_uint64(data, "view-id");
+        if (!id.has_value())
+        {
+            // Deprecated API usage: 'id' field for view is deprecated.
+            id = wf::ipc::json_get_optional_uint64(data, "id");
```

Comment:
This one shouldn't be optional, if the first one fails, then we require `id`. Same in the other places.